src/main/java/co/rsk/bitcoinj/script/Script.java

```diff
@@ -489,8 +490,51 @@ public Script getScriptSigWithSignature(Script scriptSig, byte[] sigBytes, int i
      * Returns the index where a signature by the key should be inserted. Only applicable to
      * a P2SH scriptSig.
      */
-    public int getSigInsertionIndex(Sha256Hash hash, BtcECKey signingKey) {
-        return this.redeemScriptParser.getSigInsertionIndex(hash, signingKey);
+    public int getSigInsertionIndex(Sha256Hash hashForSignature, BtcECKey signingKey) {
+        // Iterate over existing signatures, skipping the initial OP_0, the final redeem script
+        // and any placeholder OP_0 sigs.
+
+
+        // In order to affect consensus, we keep returning zero as the default value for the insertion index
+        // when the script is not a scriptSig.
+        final int defaultSigInsertionIndexForNoScriptSig = 0;
+        String noScriptSigFormatExceptionMessage = "Script does not match ScriptSig format";
+        if (chunks.size() < 2) {
+            log.debug(noScriptSigFormatExceptionMessage);
+            return defaultSigInsertionIndexForNoScriptSig;
+        }
+
+        List<ScriptChunk> chunksWithoutRedeemScript = chunks.subList(1, chunks.size() - 1);
+        ScriptChunk redeemScriptChunk = chunks.get(chunks.size() - 1);
```

Comment:
```
int redeemScriptChunkIndex = chunks.size() - 1;
ScriptChunk redeemScriptChunk = chunks.get(redeemScriptChunkIndex);

List<ScriptChunk> signaturesChunks = chunks.subList(1, redeemScriptChunkIndex);
```

wdyt?

Reply:
Looks redundant to me. By the prefix chunks, we already stated that we are talking about chunks.

Reply:
okay no worries!